Entities/MOSprite.cpp

```diff
@@ -460,8 +461,58 @@ Vector MOSprite::UnRotateOffset(const Vector &offset) const
 void MOSprite::Update()
 {
     MovableObject::Update();
+
+    ////////////////////////////////////////
+    // Animate the sprite, if applicable
+
+    if (m_FrameCount > 1)
+    {
+        if (m_SpriteAnimMode == ALWAYSLOOP)
+        {
+            float cycleTime = ((long)m_SpriteAnimTimer.GetElapsedSimTimeMS()) % m_SpriteAnimDuration;
+            m_Frame = floorf((cycleTime / (float)m_SpriteAnimDuration) * (float)m_FrameCount);
+        }
+        else if (m_SpriteAnimMode == ALWAYSRANDOM)
+        {
+            if (m_SpriteAnimTimer.GetElapsedSimTimeMS() > (m_SpriteAnimDuration / m_FrameCount))
+            {
+                //  Quick switch to other frame if only two
+                if (m_FrameCount == 2) { m_SpriteAnimMode = ALWAYSLOOP; }
+                {
+                    int prevFrame = m_Frame;
+                    // Keep trying ot find a new frame
+                    do
+                    {
+                        m_Frame = floorf((float)m_FrameCount * PosRand());
+                    } while (m_Frame == prevFrame);
+                }
+
+                m_SpriteAnimTimer.Reset();
+            }
+        }
+        else if (m_SpriteAnimMode == ALWAYSPINGPONG)
+        {
+            if (m_FrameCount == 2) { m_SpriteAnimMode = ALWAYSLOOP; }
+            int realFrameCount = m_FrameCount - 1;
```

Comment:
Get rid of this variable. You use it in one place so there's no benefit to it, and once you're more familiar with cpp you'll realize that this is just how arrays/numbers tend to work and it won't seem weird :)